pkg/client/linstor.go

```diff
@@ -611,28 +570,17 @@ func (s *Linstor) Attach(ctx context.Context, volId, node string, readOnly, rwxB
 	return nil
 }
 
-// inspectExistingResource inspects a resource to determine the right diskless
-func inspectExistingResource(resource *lapi.Resource) (*bool, string) {
+// getDisklessFlag inspects a resource to determine the right diskless flag to use.
+func getDisklessFlag(resource *lapi.Resource) string {
 	layer := &resource.LayerObject
 
-	isAvailableDiskful := true
-	isUnavailableDiskful := false
-
 	for layer != nil {
 		if layer.Type == devicelayerkind.Drbd {
```

Comment:
as this got simpler, make the `layer.Type` a switch? As is these are just 2 weird independent `if`s. At least an `else if` I'd say.

Reply:
Thanks, changed to a switch.